lib/child_process.js

```diff
@@ -535,12 +532,17 @@ function copyProcessEnvToEnv(env, name, optionEnv) {
   }
 }
 
-function normalizeSpawnArguments(file, args, options) {
-  validateString(file, 'file');
-  validateArgumentNullCheck(file, 'file');
+function validateStringParam(param, paramName) {
+  validateString(param, paramName);
+  validateArgumentNullCheck(param, paramName);
+
+  if (param.length === 0) {
+    throw new ERR_INVALID_ARG_VALUE(paramName, param, 'cannot be empty');
+  }
+}
 
-  if (file.length === 0)
-    throw new ERR_INVALID_ARG_VALUE('file', file, 'cannot be empty');
+function normalizeSpawnArguments(file, args, options) {
+  validateStringParam(file, 'file');
```

Comment:
If we keep the changes as before, the compiler has to do less inlining.

```suggestion
function normalizeSpawnArguments(file, args, options) {
  validateString(file, 'file');
  validateArgumentNullCheck(file, 'file');
  if (file.length === 0) {
    throw new ERR_INVALID_ARG_VALUE(file, 'file', 'cannot be empty');
  }
```